src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/NetworkedComponents/MeshFilter/MeshFilterBroadcaster.cs

```diff
@@ -38,9 +38,9 @@ protected override void OnInitialized()
 
             meshFilter = GetComponent<MeshFilter>();
             assetId = AssetService.Instance.GetMeshId(meshFilter.sharedMesh);
-            if (assetId == Guid.Empty)
+            if ((meshFilter.sharedMesh != null) && (assetId == Guid.Empty))

```

Comment:
>if [](start = 12, length = 2)

In the Build2019 demo app, there's one mesh filter whose mesh is missing.  That would make the MeshFilterBroadcaster log an error about the asset not being the asset cache.  This change makes it sync across null/missing meshes without treating them as errors.

Make sense?  Want me to keep this change in with my PR or revert it?

Reply:
what's the benefit of creating/syncing null/missing meshes? Are we hoping that logging an unknown mesh on both the user and spectator devices makes fixing the problem easier to troubleshoot?

Reply:
I made it so it *wouldn't* log if someone tried to sync a null/missing mesh.

Since a null mesh is a valid state, and since a missing resource is typically already warned about by Unity, it doesn't seem like SpectatorView's roll to log in either of those cases.  SpectatorView's roll should just be to make the observer see the same visuals as the broadcaster, and sync'ing over the null mesh accomplishes that.

... but in truth, I only noticed this because I was getting a confusing error message when running the Build2019Demo scene and wanted to make sure I hadn't broken something.  If y'all would prefer to keep the error, but reword it to also mention null or missing meshes, I could do that instead, and I could remove the offending MeshFilter/GameObject from the Build2019Demo scene to make the error go away instead.

Thoughts?

---
In reply to: [337250788](https://github.com/microsoft/MixedReality-SpectatorView/pull/250#discussion_r337250788) [](ancestors = 337250788)

Reply:
As noted elsewhere, I'm reverting this change, and we can consider it as part of a separate PR.